internal/logger/http_logger.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Canonical.
+package logger
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+)
+
+// HTTPLogFormatter is an implementation of chimiddleware.LogFormatter. It formats logs for http requests.
+type HTTPLogFormatter struct{}
```

Comment:
Can we have a constructor for this? We've been trying to keep to constructors recently

Reply:
we don't gain anything from a constructor with empty arguments

Reply:
agreed.. no need for a constructor.. it is an exported type. If we needed certain parameters, then, yes, constructor would make sense.